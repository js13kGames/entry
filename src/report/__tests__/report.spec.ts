import { generateReport, parseReport } from "..";

describe("report", () => {
  const bundles: Bundle[] = [
    {
      releaseName: "v1",
      deployUrl: "/a",
      releaseUrl: "/b",
      warnings: ["size is over the limit", "index not found"]
    },
    {
      releaseName: "v2",
      deployUrl: undefined,
      releaseUrl: "/d",
      warnings: ["index not found"]
    },
    {
      releaseName: "v3",
      deployUrl: "/e",
      releaseUrl: "/f",
      warnings: []
    }
  ];

  it("should generate a report", () => {
    expect(generateReport(bundles)).toMatchSnapshot();
  });

  it("should parse a report", () => {
    expect(parseReport(generateReport(bundles))).toEqual(bundles);
  });
});
