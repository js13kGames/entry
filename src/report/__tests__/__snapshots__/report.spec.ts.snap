// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`report should generate a report 1`] = `
"[v1](/b) [deployed](/a) ⚠️
 - size is over the limit
 - index not found

[v2](/d) ⚠️
 - index not found

[v3](/f) [deployed](/e) ✔️"
`;
